### PR TITLE
Move main container block to before the sidecars

### PR DESCRIPTION
**Pull request by tcnghia**

### CODE REVIEW

File: modules/regional-go-service/main.tf [hunk 1]
```diff
@@ -38,13 +38,25 @@ resource "ko_build" "this" {
...
 
+locals {
+  // Pull out the containers from var.containers that has no port
+  sidecars = {
+    for key, value in var.containers : key => value if length(value.ports) == 0
+  }
+  // Pull out the main container from var.containers that has a port.
+  // There should be only one of them, but using a map to make it easier to
+  // iterate over and look up the ko_builds.
+  main_container = {
```
Comment (k4leung4) [inline modules/regional-go-service/main.tf:55]: do we want at check to see length(main_container) == 1?

we dont want this greater than one.

Comment (tcnghia) [reply modules/regional-go-service/main.tf]: Done

File: modules/regional-go-service/main.tf [hunk 2]
```diff
@@ -77,8 +89,12 @@ resource "google_cloud_run_v2_service" "this" {
     }
     service_account = var.service_account
     timeout         = "${var.request_timeout_seconds}s"
+
+    // A main container has ports. It needs to go first to avoid a bug in the
+    // Cloud Run terraform provider where ommiting the port{} block does not
+    // remove the port from the service.
     dynamic "containers" {
-      for_each = var.containers
+      for_each = local.main_container
```
[the remark this answers is not in the document]
Comment (tcnghia) [reply modules/regional-go-service/main.tf]: Done

**Review by k4leung4 — APPROVED** [review]
thanks

**Pull request merged**